R/archive.R

```diff
@@ -313,38 +313,15 @@ new_epi_archive <- function(
     )
   }
 
-  # Checks to see if a value in a vector is LOCF
-  is_locf <- function(vec) { # nolint: object_usage_linter
-    dplyr::if_else(!is.na(vec) & !is.na(dplyr::lag(vec)),
-      vec == dplyr::lag(vec),
-      is.na(vec) & is.na(dplyr::lag(vec))
-    )
-  }
-
-  # LOCF is defined by a row where all values except for the version
-  # differ from their respective lag values
-
-  # Checks for LOCF's in a data frame
-  rm_locf <- function(df) {
-    dplyr::filter(df, if_any(c(everything(), -version), ~ !is_locf(.))) # nolint: object_usage_linter
-  }
-
-  # Keeps LOCF values, such as to be printed
-  keep_locf <- function(df) {
-    dplyr::filter(df, if_all(c(everything(), -version), ~ is_locf(.))) # nolint: object_usage_linter
-  }
-
+  nrow_before_compactify <- nrow(DT)
   # Runs compactify on data frame
   if (is.null(compactify) || compactify == TRUE) {
-    elim <- keep_locf(DT)
-    DT <- rm_locf(DT) # nolint: object_name_linter
-  } else {
-    # Create empty data frame for nrow(elim) to be 0
-    elim <- tibble::tibble()
+    DT <- compactify_tibble(DT, key_vars) # nolint: object_name_linter
```

Comment:
This is more or less a line for line replacement of `rm_locf`, with the addition of making sure that the key columns are in order via `arrange`. I'm still not sure why it was a data.table to begin with. Going to leave messing around with this to whoever/whenever we address #495 .

Reply:
For now, could you just rename `compactify_tibble` to something without `tibble` in the name, since it's being used on `data.table`s and apparently works?

Reply:
I was trying to indicate that the returned type was a tibble, but will do. there's no convention to do as I was doing.